lib/scrub/ocn.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require "scrub/common"
+require "scrub/max_ocn"
+
+module Scrub
+  class Ocn < Scrub::Common
+    # Take a string, presumed to be one or more OCNs, and get the actual OCNs out as numbers.
+    def initialize(str)
+      @ok_ocns = []
+
+      if str.nil?
+        raise ArgumentError, "ocn is nil"
+      end
+
+      # Get current max oclc
+      @current_max_ocn = Scrub::MaxOcn.new.ocn
+      if @current_max_ocn.nil?
+        raise "failed to set current_max_ocn"
+      end
+
+      count_x(:rec_with_ocn)
+      candidates = str.split(OCN_SPLIT_DELIM).map(&:strip)
```

Comment:
Could use an additional OCNCandidate class we could hide stick all the validation in, but this is pretty easy to understand as is.

Reply:
Yeah, that makes it neater. Done.